pkg/services/dashboards/dashboard_service.go

```diff
@@ -165,6 +165,7 @@ func (dr *dashboardServiceImpl) updateAlerting(cmd *models.SaveDashboardCommand,
 	}
 
 	if err := bus.Dispatch(&alertCmd); err != nil {
+		dr.log.Error("Failed to save alert data", "error", err)
```

Comment:
I think we should avoid logging and returning the same error. I expect error log to be at the level they are handled. 

Perhaps we should log all "unhandled" errors in https://github.com/grafana/grafana/blob/master/pkg/api/dashboard.go#L276

Or would that be to verbose?

Reply:
Yes that's a good point. I've never understood why `ErrDashboardFailedToUpdateAlertData` have been needed. Maybe a rest from the past? Will change.